app/SendReason.tsx

```diff
@@ -0,0 +1,189 @@
+  const router = useRouter();
+  const currentUser = useAppSelector<User>(selectUser);
+  const chosenContact = useAppSelector<Contact>(selectChosenContact);
+  const amount = useAppSelector<number>(selectAmount);
+  const reason = useAppSelector<string>(selectReason);
+  const currentUserId = useAppSelector<string>(selectUserId);
+  const dispatch = useAppDispatch();
+  const params = useLocalSearchParams();
+  const { isRequest } = params;
+
+  const onPressSend = async () => {
+    if (!reason || reasonError) {
+      setReasonError(true);
+      dispatch(setModalText('not enough funds')); // tale care of this later on
+      dispatch(setIsModalVisible(true));
+      return;
+    }
+    if (!chosenContact?.phoneNumbers) return;
+    // phone number for recipient in send flow and 'purchaser' (aka requestee) in request flow
+    const phoneNumber = chosenContact?.phoneNumbers[0].number;
+    if (!phoneNumber) return;
+    if (
+      currentUser?.accountBalance <= 0 ||
+      amount > currentUser?.accountBalance
+    ) {
+      dispatch(setModalText(i18n.t('sendamount_not_enough_funds')));
+      dispatch(setIsModalVisible(true));
+      return;
+    }
+    dispatch(setIsUserLoading(true));
+    try {
+      const parsedPhoneNumber = parsePhoneNumber(phoneNumber, 'US')
+        .nationalNumber as string;
+      if (parsedPhoneNumber === currentUser.phoneNumber) {
+        throw new Error('Cannot send or make request to yourself');
+      }
+      const { data: getUserIdByNumberData } = await dispatch(
+        getUserIdByNumber(parsedPhoneNumber)
+      ).unwrap();
+      const requesteeId = getUserIdByNumberData.userId;
+      const newTransaction = {
+        recipientPhoneNumber: isRequest
+          ? currentUser.phoneNumber
+          : parsedPhoneNumber,
+        amount: amount,
+        category: Category.MISC,
+        reason: reason,
+        // in a request the purchaserid is the id of person the request is sent to and is therefore not known
+        purchaserId: isRequest ? requesteeId : currentUserId,
+        ...(isRequest && { isRequest: true }),
+      };
+      const { data: transaction } = await dispatch(
+        saveTransaction(newTransaction)
+      ).unwrap();
+      dispatch(addUserTransaction(transaction));
+      if (!isRequest) {
+        const updatedBalance = currentUser.accountBalance - amount;
+        dispatch(setUserBalance(updatedBalance));
+      }
+      dispatch(setAmount(0));
+      dispatch(setReason(''));
+      dispatch(setIsUserLoading(false));
+      router.push('/Home');
+    } catch (e) {
+      console.error('Transaction error:', e);
+      const errorMessage =
+        e instanceof Error ? e.message : 'WRONG PHONE NUMBER';
+      dispatch(setModalText(errorMessage));
+      dispatch(setIsUserLoading(false));
+      dispatch(setIsModalVisible(true));
+    }
+  };
+
+  const onChangeReason = (reason: string) => {
+    if (!reason) {
+      setReasonError(true);
+

```

Comment:
we still need to set reason here even if reason = empty string. 
```suggestion

  dispatch(setReason(reason));
```
otherwise user can delete reason (reason=''), click back, click continue and then reason looks blank but is still present in redux store